e2e_playwright/st_markdown_test.py

```diff
@@ -105,18 +94,86 @@ def test_markdown_displays_long_headers_above_other_elements(
         app.get_by_test_id("stVerticalBlock").get_by_test_id("stVerticalBlock").nth(0)
     )
 
-    assert_snapshot(long_header, name="st_markdown-long-markdown-header-above-table")
+    assert_snapshot(long_header, name="st_markdown-header_long_above_markdown_table")
+
+
+def _get_container_of_text(app: Page, text: str) -> Locator:
+    """Get the parent container in which the passed text is located.
+    The tests are written in a way that the text and the headers are put
+    into the same container.
+    """
+
+    # take the 2nd match because the first would be the most outer block
+    return (
+        app.get_by_test_id("stVerticalBlock")
```

Comment:
clever :) But would be even simpler if we could add a class to a container via key. Hopefully, we can add that soon.

Reply:
yes!